### DISCUSSION TITLE
debugging process

### DISCUSSION THREAD

**Discussion by khm** [tvler/streetpass#61] [Q&A] [answered]
Hello,

I'm trying to ascertain why Streetpass isn't triggering on my site, and I'm not sure what the best process is for this.  Running the querySelectorAll code does return a NodeList containing the expected _a_ element, but the extension never triggers.  The site I'm testing with is https://sciops.net -- any advice on how to trace what's supposed to happen vs. what actually happens would be appreciated.

**Comment by tvler** [comment, marked as answer]
Hi thank you for reaching out! It looks like it's because this webfinger response (https://hj.9fs.net/.well-known/webfinger?resource=https://hj.9fs.net/khm) isn't returning a `http://webfinger.net/rel/profile-page` link. Streetpass doesn't return a profile if that link isn't in the webfinger resp: https://github.com/tvler/streetpass/blob/main/extension/src/util/getUncachedProfileData.ts#L81

This is because it's not guaranteed that the `application/activity+json` link will link to an actual human-readable webpage IIRC. So the solution would be to keep what you have and add a new `http://webfinger.net/rel/profile-page` link to it!
<img width="331" alt="image" src="https://github.com/tvler/streetpass/assets/4934193/5211a175-4548-4fb0-8585-d2f89bb0fd10">

Here's an example of a working webfinger response https://mastodon.social/.well-known/webfinger?resource=acct:tvler@mastodon.social

**Reply by tvler** [reply]
Once you fix that, wait 10 minutes for the cache to run out and streetpass will be able to pick it up! https://github.com/tvler/streetpass/blob/00b9e02879b11a2b9eb4cd14705b5e702dfddba2/extension/src/util/constants.ts#L85

**Reply by khm** [reply]
I added this output and it works great now.  Can I ask how you diagnosed this?  Would it be worth logging validation failures to console or similar?

Anyway, I opened a pull request for the ActivityPub software project I'm using to get this fixed for good.

**Reply by tvler** [reply]
> Can I ask how you diagnosed this

Sure! First I found the webfinger response that your instance was returning. Then I went through the `getUncachedProfileData` (https://github.com/tvler/streetpass/blob/main/extension/src/util/getUncachedProfileData.ts) function and just kinda pattern matched seeing what would happen if your webfinger response was passed into that. That function's really the only point of failure for the issue you were seeing :)

Definitely down to make debugging easier! That's a goal of mine (along with tests) for streetpass this year

**Reply by khm** [reply]
Thanks for the info!  Appreciate your help.  The webfinger change has been merged upstream, so the next release of snac2 will work with Streetpass.

**Reply by tvler** [reply]
Exciting! glad I could help
